tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -2264,7 +2264,8 @@ bool DeclResultIdMapper::createStageVars(
     // Decorate with interpolation modes for pixel shader input variables
     // or vertex shader output variables.
     if (((spvContext.isPS() && sigPoint->IsInput()) ||
-         (spvContext.isVS() && sigPoint->IsOutput())) &&
+         (spvContext.isVS() && sigPoint->IsOutput()) ||
+         (spvContext.isHS() || spvContext.isDS())) &&
```

Comment:
I'm curious why this is needed? (why `Flat` should be applied)

Reply:
This code is from the UnrealEngine's code, but I do not think this is needed.
I removed it.

Reply:
@LukasBanana Do you know why this change was made in the UE fork? it results in 

```
OpDecorate %gl_InvocationID Flat
OpDecorate %gl_PrimitiveID Flat 
```
being added to hull shaders and domain shaders.

Reply:
I added that change myself as I was assuming all integer types need the flat interpolation specifier for stage variables in SPIRV, except of vertex input and fragment output attributes of course. Now that I think about it, it's probably only required for the variables that are in fact interpolated, i.e. for the fragment shader. But if that's the case, why would it be required in the vertex shader (i.e. `spvContext.isVS() && sigPoint->IsOutput()`) if a geometry shader for instance is hooked into the pipeline? In this scenario, the vertex shader wouldn't interpolate its vertices, the geometry shader would do that.

Reply:
Thanks 👍